src/runner/PlutoRunner.jl

```diff
@@ -683,4 +684,36 @@ const fake_bind = """macro bind(def, element)
     end
 end"""
 
+###
+# LOGGING
+###
+
+const log_channel = Channel{Any}(10)
+
+struct PlutoLogger <: Logging.AbstractLogger end
+
+const to_pluto_html(@nospecialize(x)) = sprint() do io
+    show_richest(io, x; onlyhtml=true)
+end
+
+Logging.shouldlog(::PlutoLogger, args...) = true
+Logging.min_enabled_level(::PlutoLogger) = Logging.Debug
+Logging.catch_exceptions(::PlutoLogger) = false
+function Logging.handle_message(::PlutoLogger, level, msg, _module, group, id, file, line; kwargs...)
+    put!(log_channel, (
+        level=string(level),
+        msg=(msg isa String) ? msg : to_pluto_html(msg),
+        group=group,
+        # id=id,
+        file=file,
+        line=line,
+        kwargs=Dict((k=>to_pluto_html(v) for (k,v) in kwargs)...),
+    ))
+end
+
+# we put this in __init__ to fix a world age problem
+function __init__()
+    Logging.global_logger(PlutoLogger())
```

Comment:
It may be better just to use `Logging.with_logger()` in some context which surrounds the user's code?